### PR TITLE
Documentation for buffer library

### PR DESCRIPTION
**Pull request by vegorov-rbx**
[Rendered](https://vegorov-rbx.github.io/site/library#buffer-library)

### CODE REVIEW

File: _pages/library.md [hunk 1]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+
+```
+function buffer.fill(b: buffer, offset: number, value: number, count: number?): ()
+```
+
+Sets the 'count' bytes in the buffer starting at the specified 'offset' to the 'value'.
+
+Offset should be a positive integer number and the value starts at 0.
+
+If 'count' is 'nil' or is omitted, all bytes after the specified offset are set.
```
Comment (zeux) [inline _pages/library.md:969]:
```suggestion
If 'count' is 'nil' or is omitted, all bytes from the specified offset until the end of the buffer are set.
```

File: _pages/library.md [hunk 2]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+
+Used to read a string of length 'count' from the buffer at specified offset.
+
+If the bytes that are accessed by the read are outside the buffer memory, an error is thrown.
+
+```
+function buffer.writestring(b: buffer, offset: number, value: string, count: number?): ()
+```
+
+Used to write data from a string into the buffer at specified offset.
```
Comment (zeux) [inline _pages/library.md:937]: at a?

File: _pages/library.md [hunk 3]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+
+Offset should be a positive integer number and the value starts at 0.
+
+If an optional 'count' is specified, only 'count' bytes are taken from the string.
+
+Count should be a positive integer number and cannot be larger than the string length.
```
Comment (zeux) [inline _pages/library.md:943]: count can also be 0

Comment (zeux) [reply _pages/library.md:943]: (this entire sentence might be redundant anyway)

File: _pages/library.md [hunk 4]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+function buffer.writeu32(b: buffer, offset: number, value: number): ()
+function buffer.writef32(b: buffer, offset: number, value: number): ()
+function buffer.writef64(b: buffer, offset: number, value: number): ()
+```
+
+Used to write data to the buffer by converting the number into the type in the argument and reinterpreting it as individual bytes.
+
+If the bytes that are accessed by the write are outside the buffer memory, an error is thrown.
+
+Offset should be a positive integer number and the value starts at 0.
```
Comment (zeux) [inline _pages/library.md:921]: offset can be 0

File: _pages/library.md [hunk 5]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+function buffer.tostring(b: buffer): string
+```
+
+Returns the buffer data as a string.
+
+```
+function buffer.len(b: buffer): number
+```
+
+Returns the size of the buffer.
```
Comment (zeux) [inline _pages/library.md:866]:
```suggestion
Returns the size of the buffer in bytes.
```

File: _pages/library.md (hunk 4 above)
Comment (vegorov-rbx) [reply _pages/library.md]: Fixed.

File: _pages/library.md [hunk 6]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+
+All operations on a buffer are provided using the 'buffer' library functions.
+
+```
+function buffer.create(size: number): buffer
+```
+
+Creates a buffer of the requested size with all bytes initialized to 0.
+
+Size should be an integer number and cannot be negative.
```
Comment (zeux) [inline _pages/library.md:843]: I would probably recommend noting offset/count/size restrictions once for all functions in the preamble above. Sizes should be non-negative integers, all buffer accesses outside of buffer bounds result in an error, etc.?

File: _pages/library.md (hunk 2 above)
Comment (vegorov-rbx) [reply _pages/library.md]: Fixed.

File: _pages/library.md (hunk 6 above)
Comment (vegorov-rbx) [reply _pages/library.md]: Added.

File: _pages/library.md [hunk 7]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+function buffer.writeu8(b: buffer, offset: number, value: number): ()
+function buffer.writei16(b: buffer, offset: number, value: number): ()
+function buffer.writeu16(b: buffer, offset: number, value: number): ()
+function buffer.writei32(b: buffer, offset: number, value: number): ()
+function buffer.writeu32(b: buffer, offset: number, value: number): ()
+function buffer.writef32(b: buffer, offset: number, value: number): ()
+function buffer.writef64(b: buffer, offset: number, value: number): ()
+```
+
+Used to write data to the buffer by converting the number into the type in the argument and reinterpreting it as individual bytes.
```
[the remark this answers is not in the document]
Comment (vegorov-rbx) [reply _pages/library.md:918]: Added.

File: _pages/library.md [hunk 8]
```diff
@@ -827,3 +827,143 @@ function debug.traceback(msg: string?, level: number?): string
...
+Offset should be a positive integer number and the value starts at 0.
+
+Available types:
+
+Function | Type | Range |
+---------|------|-------|
+readi8 | signed 8-bit integer | [-128, 127]
+readu8 | unsigned 8-bit integer | [0, 255]
+readi16 | signed 16-bit integer | [-32,768, 32,767]
+readu16 | unsigned 16-bit integer | [0, 65,536]
```
[the remark this answers is not in the document]
Comment (vegorov-rbx) [reply _pages/library.md:893]: It had to happen eventually.

File: _pages/library.md [hunk 9]
```diff
@@ -827,3 +827,140 @@ function debug.traceback(msg: string?, level: number?): string
...
+
+Results of converting special number values (inf/nan) to integers are platform-specific.
+
+```
+function buffer.readstring(b: buffer, offset: number, count: number): string
+```
+
+Used to read a string of length 'count' from the buffer at specified offset.
+
+If the bytes that are accessed by the read are outside the buffer memory, an error is thrown.
```
Comment (zeux) [inline _pages/library.md:936]:
```suggestion

```

**Pull request merged**